gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/ApisResource.java

```diff
@@ -503,6 +513,7 @@ public PagedResult<ApiListItem> searchPagedApis(
     @ApiResponse(responseCode = "403", description = "Forbidden")
     @ApiResponse(responseCode = "500", description = "Internal server error")
     public Response updateWithDefinition(@RequestBody(required = true) @Valid @NotNull Object apiDefinitionOrUrl) {
+        // TODO: how to check definition version ?
```

Comment:
Maybe in the service impl once it is converted to UpdateApiEntity? Can throw an exception related to "v1" not allowed.

Reply:
Following my previous comment, if you move all checks to the services, you will be able to do it after the definition is resolved.